packages/core/src/baseclient.ts

```diff
@@ -655,6 +657,13 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
     );
   }
 
+  /**
+   * Loads attachment items from scope
+   */
+  protected _attachmentsFromScope(scope: Scope | undefined): AttachmentItem[] | undefined {
+    return scope?.getAttachments()?.map(a => createAttachmentEnvelopeItem(a));
```

Comment:
What do you think about changing `_attachmentsFromScope` so that it always returns an array? I think this better matches with the underlying scope getter.